File: frontend/src/types/category.ts
export interface CreateCategoryDto{
     name:string;
     description:string;
     interviewPrompt:string;
     evaluationPrompt:string;
     defaultDifficulty:number;
     maxQuestions:number;
     isActive:boolean;
}
export interface CategoryDto{
     id:string;
     name:string;
     description:string;
     interviewPrompt:string;
     evaluationPrompt:string;
     defaultDifficulty:number;
     maxQuestions:number;
     isActive:boolean;
     createdAt: Date;
}
export interface ChangeCategoryDto{
     id:string;
     name:string;
     description:string;
     interviewPrompt:string;
     evaluationPrompt:string;
     defaultDifficulty:number;
     maxQuestions:number;
     isActive:boolean;
}

File: backend/AInterviewer/AInterviewer/DTOs/Category.cs
namespace AInterviewer.DTOs;

public record CreateCategoryDto(
    string Name,
    string Description,
    string InterviewPrompt,
    string EvaluationPrompt,
    int DefaultDifficulty,
    int MaxQuestions,
    bool IsActive
);
public record CategoryDto(
    Guid Id,
    string Name,
    string Description,
    string InterviewPrompt,
    string EvaluationPrompt,
    int DefaultDifficulty,
    int MaxQuestions,
    bool IsActive,
    DateTime CreatedAt
);
public record ChangeCategoryDto(
    Guid Id,
    string Name,
    string Description,
    string InterviewPrompt,
    string EvaluationPrompt,
    int DefaultDifficulty,
    int MaxQuestions,
    bool IsActive
);